.github/workflows/commitint.yml

```diff
@@ -0,0 +1,23 @@
+name: commitlint
```

Comment:
i actually think it'd probably be ever-so-slightly better to put this as a separate job in `ci.yml` because it is indeed another ci step. we can consider splitting it into a separate file if `ci.yml` gets to be big.

Reply:
perfect sounds good to me!